packages/clap-host/PKGBUILD

```diff
@@ -0,0 +1,48 @@
+# Maintainer: OSAMC <https://github.com/osam-cologne/archlinux-proaudio>
+# Contributor: Florian Hülsmann <fh@cbix.de>
+
+pkgname=clap-host
+pkgver=1.0.2
+pkgrel=1
+pkgdesc='CLAP example host'
+arch=(aarch64 x86_64)
+url='https://github.com/free-audio/clap-host'
+license=(MIT)
+depends=(gcc-libs qt6-base)
+makedepends=(catch2 cmake rtaudio rtmidi)
+groups=(pro-audio)
+_clap_ref='27b7af0'
+_clap_helpers_ref='4d4e297'
+source=("$pkgname-$pkgver.tar.gz::https://github.com/free-audio/$pkgname/archive/refs/tags/$pkgver.tar.gz"
+        "clap-$_clap_ref.tar.gz::https://github.com/free-audio/clap/archive/$_clap_ref.tar.gz"
+        "clap-helpers-$_clap_helpers_ref.tar.gz::https://github.com/free-audio/clap-helpers/archive/$_clap_helpers_ref.tar.gz")
+sha256sums=('ec4342d8edcdd8262da1042cd6b3862d180dfaf740eb326239b86cbb6331dd4e'
+            '8a85ac0fe0ce58c6e641386257a21bd5167b55f05bdd0e69b9617c49b78b9a6b'
+            '0681e844336212e56df41b9ed7937b6e0d82f4005209e392fbc991e5d3ed6dd4')
+noextract=(clap-$_clap_ref.tar.gz clap-helpers-$_clap_helpers_ref.tar.gz)
+
+prepare() {
+  cd $pkgname-$pkgver
+  bsdtar -xf ../clap-$_clap_ref.tar.gz -C clap --strip-components 1
+  bsdtar -xf ../clap-helpers-$_clap_helpers_ref.tar.gz -C clap-helpers --strip-components 1
+}
+
+build() {
+  cmake -B build -S $pkgname-$pkgver \
+    -DCMAKE_BUILD_TYPE=None \
+    -DCMAKE_INSTALL_PREFIX=/usr \
+    -DCLAP_BUILD_TESTS=On \
+    -Wno-dev
+  cmake --build build --target clap-host clap-tests
+}
+
+check() {
+  ctest --test-dir build --output-on-failure
+}
+
+package() {
+  depends+=(hicolor-icon-theme librtaudio.so librtmidi.so)
```

Comment:
Why not put `hicolor-icon-theme` in the `depends` list in the header?

Reply:
It's not required at build time, only when installing.

Reply:
Ok, I see. But I wonder why I never see t done that way in the official repos.

One disadvantage is that wrong dependencies will only be caught when actually installing the package, or is it not?

Reply:
`hicolor-icon-theme` is a special package as it only includes the necessary directory structure for the hicolor icon theme, so desktop managers can use that tree to look for icons. Lots of packages install their application icon in that tree, so in order to make the icon "discoverable", they need `hicolor-icon-theme` as a dependency. Also, namcap has a check for that :D